### ISSUE TITLE
write spec for use of message fields

### ISSUE THREAD

**Issue by aelkiss** [hathitrust/ht_sip_validator#72]
Figure out a use case for the content of the `extras` fields would be used for as well as:

- `validation` (which should have a controlled vocabulary in the future and needs to be renamed?)
- `validator` (class of validator)

If we have no use cases, we should not be putting those in the messages, or at least not caring about what goes in there.

**Comment by malakai97** [comment]
These fields make the messages machine readable, so use cases I can think of:

1. Automatic remediation based on messages.
2. Output formatters

**Comment by aelkiss** [comment]
Yeah, the automatic remediation based on messages is a good one, since for the image validation we were thinking of having the image validation validate to stricter specs but then something else that filters out problems known to be automatically remediable. I think that would be a good use case for field, file, and `validation`. 

So, fields and use cases:

- `validation`: automatic remediation, maybe display; I have a hard time seeing utility for filtering for this one. In the current ingest code I have a similar field (basically 'error type') where sometimes I filter a database for things like 'Unexpected Error' or for something like 'Field Missing' with a given field. That might have less utility here, but I could maybe see it if people were validating a large number of packages at once.

- `validator`: automatic remediation, output filtering, display
- `filename`: automatic remediation, output filtering, display
- `field`: automatic remediation, output filtering, display
- `actual`: display, perhaps remediation, perhaps output filtering (for large scale validation)
- `expected`: maybe display
- `root_cause` (error returned by underlying parser): display, perhaps remediation
